Makefile-test.am

```diff
@@ -16,9 +16,11 @@ if TESTPTPM
 INT_LOG_COMPILER = $(srcdir)/script/int-log-compiler-ptpm.sh
 INT_LOG_FLAGS = --ptpm=$(PTPM)
 else
-INT_LOG_COMPILER = $(srcdir)/script/int-log-compiler.sh
+INT_LOG_COMPILER = $(srcdir)/script/int-log-compiler-swtpm.sh
```

Comment:
As long as we still support tpm_server, it might be nice to use `AC_CHECK_PROG` whether tpm_server or swtpm are available and switch the log compiler accordingly. Depends a bit on how long we want to keep tpm_server around whether it's worth the effort...

Reply:
Good call. I added  a mechanism which chooses the testing backend. I also streamlined the code for choosing `device` (formerly `ptpm`) with this new mechanism.

In configure.ac, I added two output variables to configure.ac:

  - INTEGRATION_TCTI: `mssim`, `swtpm` or `device`   
(according to this variable, int-log-compiler-<...>.sh will be called for testing)
  - INTEGRATION_ARGS: arguments to passed to int-log-compiler script   
(`--device=<...>` for device, empty otherwise)

 The mechanism for choosing the testing backend is as follows:

  - if --with-device=<..> was given:
     - INTEGRATION_TCTI = `device`
     - INTEGRATION_ARGS = `--device=<..>`

  - if tcti-swtpm is enabled and swtpm in PATH
     - INTEGRATION_TCTI = `swtpm`

  - if tcti-mssim is enabled and tpm_server in PATH
     - INTEGRATION_TCTI = `mssim`

   - if no backend could be found: error